pipeline.lisp

```diff
@@ -15,28 +15,64 @@
    (%names :initarg :names :initform (make-hash-table :test 'equal) :accessor names))
   (:documentation "Base pipeline object that is necessary to orchestrate piping systems."))
 
+(define-condition unknown-name-error ()
+  ((%name :initarg :name :reader name)
+   (%pipeline :initarg :pipeline :reader pipeline))
+  (:documentation "Signalled if a name which was not previously set in the pipeline is looked up")
+  (:report (lambda (c stream)
+             (format stream "Name ~A is not known in pipeline ~A" (name c) (pipeline c)))))
+
+(defgeneric resolve-place (pipeline place &key ignore-unknown)
+  (:documentation "Resolve PLACE to a path (ie. list of integers). If PLACE is already a
+  path, return it unchanged. If it's a name (as set by SET-NAME), return its corresponding
+  path. If the name is not known, signal UNKNOWN-NAME-ERROR, unless IGNORE-UNKNOWN is
+  non-nil, in which case NIL is returned.
+
+  Returns a secondary value which is T if the place was successfully resolved, and NIL if
+  it was an unknown name and INGORE-UNKNOWN was specified.")
+  (:method ((pipeline pipeline) (place list) &key &allow-other-keys)
+    (values place t))
```

Comment:
`t` and `nil` are constants and should thus be uppercase: `T` `NIL`.

Reply:
You mean in the code? Cause they already are in the docstring. I've never seen anyone do it that way in the code, and in fact that would be harmful for if the readtable-case is `:invert`, [which is very useful](https://www.cliki.net/case%20sensitivity) and I've been trying to get libraries usable with it for a long time.

Reply:
Yes in the code. This is how it is in all my code and libraries, and it will continue to be so.

Reply:
Besides, messing with the readtable case of libraries is just as unsupported as it would be to set `*read-base*` to anything but `10`.

Reply:
It's only "funny business" because people assume `:UPCASE`, thus making anything else unusable, but sure. Your code already follows that.